File: wp2db.py
import csv
# Title,Code,Country,Latitude,Longitude,Elevation,Style,Direction,Length,Frequency,Description
with open('wp.txt', 'rb') as csvfile:
     reader = csv.reader(csvfile, delimiter=',', quotechar='"')
     first=True
     for row in reader:
         #print ': '.join(row)
	 if first:
		first=False
		continue
     	 title 		= row[0]
     	 code 		= row[1]
     	 country 	= row[2]
     	 lati 		= row[3]
	 latitude = float(lati[0:8])/100
	 if lati[8] =='S':
		latitude *= -1
     	 long		= row[4]
	 longitude= float(long[0:9])/100
	 if long[9] == 'W':
		longitude *= -1
     	 alti		= row[5]
     	 if alti[-1] == 'm':
		altitude =int(alti[0:-1])
     	 style 		= row[6]
	 print title, code, country, latitude, longitude, altitude, style

File: wp2db.php
<?php
require_once  'config.php';
// Create connection
$connAPRSLOG = new mysqli($servername, $DBuser, $DBpasswd, $dbname);

$fila = 1;
if (($gestor = fopen("wp.txt", "r")) !== FALSE) {
    while (($row = fgetcsv($gestor, 1000, ",")) !== FALSE) {
	if ($fila == 1){
		$fila++;
		continue;
		}
        $numero 	= count($row);
        $fila++;
     	$title 		= $row[0];
     	$code 		= $row[1];
     	$country 	= $row[2];
     	$lati 		= $row[3];
	$latitude = floatval(substr($lati,0,2))+floatval(substr($lati,2,6))/60;
	if ($lati{8} =='S'){
		$latitude *= -1;
		}
     	$long		= $row[4];
	$longitude= floatval(substr($long,0,3))+floatval(substr($long,3,6))/60;
	if ($long{9} == 'W'){
		$longitude *= -1;
		}
     	$alti		= $row[5];
	$altitude =intval(substr_replace($alti,"m",0));
     	$style 		= $row[6];
	//echo $title, " ", $code," ",  $country," ",  $latitude," ",  $longitude," ",  $altitude," ",  $style, "\n";
	$sqlString="INSERT INTO WAYPOINTS (idWaypoint, waypoint, waypointType, waypointCountry, waypointLat, waypointLon) values ('$fila','".$title."','".$style."','".$country."','".$latitude."','".$longitude."');";	
	$result = $connAPRSLOG->query($sqlString);	
	echo $sqlString, $result, "\n";
    }
    fclose($gestor);
}
$connAPRSLOG->close();

?>

